### PR TITLE
Create C# inference section and added C# BERT NLP tutorial

### PR DESCRIPTION
**Pull request by cassiebreviu**
Created a new section for the C# tutorials that were not previously part of the navigation. Added a new tutorial on how to use BERT and C#.

https://cassiebreviu.github.io/onnxruntime/docs/tutorials/csharp/

### CODE REVIEW

File: docs/tutorials/csharp/basic_csharp.md [hunk 1]
```diff
@@ -1,7 +1,6 @@
 ---
 nav_exclude: true 
 ---
-
```
Comment (natke) [inline docs/tutorials/csharp/basic_csharp.md:4]: Should this one be in the nav too?

File: docs/tutorials/csharp/bert-nlp-csharp-console-app.md [hunk 2]
```diff
@@ -0,0 +1,376 @@
+---
+title: Inference with C# BERT NLP and ONNX Runtime
```
Comment (natke) [inline docs/tutorials/csharp/bert-nlp-csharp-console-app.md:2]: This is where I would add "deep learning" to the meta title and meta description

File: docs/tutorials/csharp/bert-nlp-csharp-console-app.md [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+grand_parent: Tutorials
+has_children: false
+nav_order: 1
+---
+
+
+# Inference with C# BERT NLP Deep Learning and ONNX Runtime
+{: .no_toc }
+
+In this tutorial we will learn how to do inferencing for the popular BERT Natural Language Processing model in C#.
```
Comment (natke) [inline docs/tutorials/csharp/bert-nlp-csharp-console-app.md:14]: Could also add "deep learning" here

File: docs/tutorials/csharp/bert-nlp-csharp-console-app.md [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+- [Visual Studio](https://visualstudio.microsoft.com/downloads/)
+- [VS Code](https://code.visualstudio.com/Download) with the [Jupyter notebook extension](https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter).
+- [Anacaonda](https://www.anaconda.com/)
+
+To run in the cloud with Azure Machine Learning:
+
+- [Azure Subscription](https://azure.microsoft.com/free/)
+- [Azure Machine Learning Resource](https://azure.microsoft.com/services/machine-learning/)
+
+## Use Hugging Face to download the BERT model
```
Comment (natke) [inline docs/tutorials/csharp/bert-nlp-csharp-console-app.md:47]: This is all great

File: docs/tutorials/csharp/bert-nlp-csharp-console-app.md [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+                                'segment_ids' : symbolic_names,
+                                'start_logits' : symbolic_names, 
+                                'end_logits': symbolic_names})   # variable length axes/dynamic input
+```
+## Understanding the model in Python
+When taking a prebuilt model and operationalizing it, its useful to take a moment and understand the models pre and post processing, and the input/output shapes and labels. Many models have sample code provided in Python. We will be inferencing our model with C# but first lets test it and see how its done in Python. This will help us with our C# logic in the next step.
+
+- Create the `preprocess`, `postprocess`, `init` and `run` functions in python to test the model.
+
+```python
```
Comment (natke) [inline docs/tutorials/csharp/bert-nlp-csharp-console-app.md:115]: Since this code is (mostly?) copied from https://onnxruntime.ai/docs/tutorials/azureml.html, I would link to it here rather than copy. It will help with the focus of your tutorial, which is C# and also do better with SEO (duplicated content tends to harm perf of both pages)

File: docs/tutorials/csharp/bert-nlp-csharp-console-app.md [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+```csharp
+    public class BertInput
+    {
+        public long[] InputIds { get; set; }
+        public long[] AttentionMask { get; set; }
+        public long[] TypeIds { get; set; }
+    }
+```
+### Tokenize the sentence with the `BertUncasedLargeTokenizer`
+- Create a sentence (question and context) and tokenize the sentence with the `BertUncasedLargeTokenizer`. The base model for this fine tuned model was the BERT Uncased Large so the tokenizer is the same.
```
Comment (natke) [inline docs/tutorials/csharp/bert-nlp-csharp-console-app.md:265]: This is not super clear. Would all fine tuned BERT models use this tokenizer? If so, just say that. If not, maybe point when you would use this and when you wouldn't

File: docs/tutorials/csharp/bert-nlp-csharp-console-app.md [hunk 7]
```diff
@@ -0,0 +1,376 @@
...
+- Create the `InferenceSession`, run the inference and print out the result.
+
+```csharp
+  // Create an InferenceSession from the Model Path.
+  var session = new InferenceSession(modelPath);
+
+  // Run session and send the input data in to get inference output. 
+  var output = session.Run(input);
+```
+### Postprocess the `output` and print the result
```
Comment (natke) [inline docs/tutorials/csharp/bert-nlp-csharp-console-app.md:342]: Maybe describe what is happening here in 1-2 sentences

Comment (natke) [reply docs/tutorials/csharp/bert-nlp-csharp-console-app.md:342]: Or link back to the other tutorial

**Review by natke — COMMENTED** [review]
Really nice! A few minor comments

File: docs/tutorials/csharp/basic_csharp.md (hunk 1 above)
Comment (cassiebreviu) [reply docs/tutorials/csharp/basic_csharp.md:4]: It didn't look complete and there is already the get started with csharp, so this seemed like a duplicate, but I didn't want to delete it. Should it be added to the get started section? Or if you think it makes sense to add it in this section, I am good with that too!

File: docs/tutorials/csharp/bert-nlp-csharp-console-app.md (hunk 2 above)
Comment (cassiebreviu) [reply docs/tutorials/csharp/bert-nlp-csharp-console-app.md:2]: Added. I was concerned about a longer title in the nav. I will see how it looks in the staged site once I check this change in.

File: docs/tutorials/csharp/bert-nlp-csharp-console-app.md (hunk 3 above)
Comment (cassiebreviu) [reply docs/tutorials/csharp/bert-nlp-csharp-console-app.md:14]: added!

File: docs/tutorials/csharp/bert-nlp-csharp-console-app.md (hunk 5 above)
Comment (cassiebreviu) [reply docs/tutorials/csharp/bert-nlp-csharp-console-app.md]: I was wondering that! Updated!

File: docs/tutorials/csharp/bert-nlp-csharp-console-app.md (hunk 6 above)
Comment (cassiebreviu) [reply docs/tutorials/csharp/bert-nlp-csharp-console-app.md:182]: There are different base models and tokenizers that a BERT model may have originated from. Based on what the model was fine tuned from indicates with tokenizer to use. I explain that a bit up earlier in the tutorial on line 25.

File: docs/tutorials/csharp/bert-nlp-csharp-console-app.md (hunk 7 above)
Comment (cassiebreviu) [reply docs/tutorials/csharp/bert-nlp-csharp-console-app.md:260]: Added!

File: docs/tutorials/csharp/bert-nlp-csharp-console-app.md (hunk 6 above)
Comment (natke) [reply docs/tutorials/csharp/bert-nlp-csharp-console-app.md:183]: So would something like: 

`Create a sentence (question and context) and tokenize the sentence with the `BertUncasedLargeTokenizer`. This tokenizer can used for all BERT based HuggingFace models.`

File: docs/tutorials/csharp/bert-nlp-csharp-console-app.md (hunk 2 above)
Comment (natke) [reply docs/tutorials/csharp/bert-nlp-csharp-console-app.md]: That's fair

File: docs/tutorials/csharp/basic_csharp.md (hunk 1 above)
Comment (natke) [reply docs/tutorials/csharp/basic_csharp.md:4]: This was has had so few views, I think we can safely delete it

Comment (natke) [reply docs/tutorials/csharp/basic_csharp.md:4]: As discussed will and pull in the content later

**Review by natke — APPROVED** [review]
